src/components/AddPaymentCard/PaymentCardForm.tsx

```diff
@@ -0,0 +1,321 @@
+            addressZipCode: 'addDebitCardPage.error.addressZipCode',
+        },
+    },
+    [CARD_TYPES.PAYMENT_CARD]: {
+        defaults: {
+            cardNumber: 'addPaymentCardPage.paymentCardNumber',
+            nameOnCard: 'addPaymentCardPage.nameOnCard',
+            expirationDate: 'addPaymentCardPage.expirationDate',
+            expiration: 'addPaymentCardPage.expiration',
+            securityCode: 'addPaymentCardPage.cvv',
+            billingAddress: 'addPaymentCardPage.billingAddress',
+        },
+        error: {
+            nameOnCard: 'addPaymentCardPage.error.invalidName',
+            cardNumber: 'addPaymentCardPage.error.paymentCardNumber',
+            expirationDate: 'addPaymentCardPage.error.expirationDate',
+            securityCode: 'addPaymentCardPage.error.securityCode',
+            addressStreet: 'addPaymentCardPage.error.addressStreet',
+            addressZipCode: 'addPaymentCardPage.error.addressZipCode',
+        },
+    },
+};
+
+function PaymentCardForm({
+    shouldShowPaymentCardForm,
+    addPaymentCard,
+    showAcceptTerms,
+    showAddressField,
+    showCurrencyField,
+    isDebitCard,
+    submitButtonText,
+    showStateSelector,
+    footerContent,
+    headerContent,
+}: PaymentCardFormProps) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+    const route = useRoute();
+    const label = CARD_LABELS[isDebitCard ? CARD_TYPES.DEBIT_CARD : CARD_TYPES.PAYMENT_CARD];
+
+    const cardNumberRef = useRef<AnimatedTextInputRef>(null);
+
+    const [isCurrencyModalVisible, setIsCurrencyModalVisible] = useState(false);
+    const [currency, setCurrency] = useState<keyof typeof CONST.CURRENCY>(CONST.CURRENCY.USD);
+
+    const validate = (formValues: FormOnyxValues<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM>): FormInputErrors<typeof ONYXKEYS.FORMS.ADD_DEBIT_CARD_FORM> => {
+        const errors = ValidationUtils.getFieldRequiredErrors(formValues, REQUIRED_FIELDS);
+
+        if (formValues.nameOnCard && !ValidationUtils.isValidLegalName(formValues.nameOnCard)) {
+            errors.nameOnCard = label.error.nameOnCard;
+        }
+
+        if (formValues.cardNumber && !ValidationUtils.isValidDebitCard(formValues.cardNumber.replace(/ /g, ''))) {
+            errors.cardNumber = label.error.cardNumber;
+        }
+
+        if (formValues.expirationDate && !ValidationUtils.isValidExpirationDate(formValues.expirationDate)) {
+            errors.expirationDate = label.error.expirationDate;
+        }
+
+        if (formValues.securityCode && !ValidationUtils.isValidSecurityCode(formValues.securityCode)) {
+            errors.securityCode = label.error.securityCode;
+        }
+
+        if (formValues.addressStreet && !ValidationUtils.isValidAddress(formValues.addressStreet)) {
+            errors.addressStreet = label.error.addressStreet;
+        }
+
+        if (formValues.addressZipCode && !ValidationUtils.isValidZipCode(formValues.addressZipCode)) {
```

Comment:
This validation logic caused the following issue:
- https://github.com/Expensify/App/issues/52642

since it was only passing US format Zip Codes. We fixed the issue by removing the validation entirely.